resources/page/permalinks.go

```diff
@@ -263,3 +272,80 @@ func (l PermalinkExpander) pageToPermalinkSection(p Page, _ string) (string, err
 func (l PermalinkExpander) pageToPermalinkSections(p Page, _ string) (string, error) {
 	return p.CurrentSection().SectionsPath(), nil
 }
+
+var (
+	nilSliceFunc = func(s []string) []string {
+		return nil
+	}
+	allSliceFunc = func(s []string) []string {
+		return s
+	}
+)
+
+func (l PermalinkExpander) toSliceFunc(cut string) func(s []string) []string {
```

Comment:
A godoc comment for this function would be helpful.  Mainly would like to know that it's parsing the `[X:Y]` slice syntax with the special `last` keyword.